rmw_fastrtps_shared_cpp/src/custom_publisher_info.cpp

```diff
@@ -160,6 +226,41 @@ void PubListener::set_on_new_event_callback(
   }
 }
 
+void
+PubListener::on_publication_matched(
+  eprosima::fastdds::dds::DataWriter * /* writer */,
+  const eprosima::fastdds::dds::PublicationMatchedStatus & info)
+{
+  rmw_event_type_t event_type = RMW_EVENT_INVALID;
+  {
+    std::lock_guard<std::mutex> lock(discovery_m_);
+    if (info.current_count_change == 1) {
+      subscriptions_.insert(eprosima::fastrtps::rtps::iHandle2GUID(info.last_subscription_handle));
```

Comment:
I know this is old, but what's being `subscriptions_` used for?
It seems that we're filling it but never using it.
In that case, I would suggest removing the container.

Reply:
`subscriptions_` is used for public API

https://github.com/ros2/rmw_fastrtps/blob/rolling/rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_publisher_info.hpp#L102-L106

And this API is called by rmw_publisher_count_matched_subscriptions()

Reply:
fair enough.
It would be better to keep a count in the future though, which we're already doing in `matched_status_`.